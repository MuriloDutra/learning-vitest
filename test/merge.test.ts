import { expect, test } from "vitest";
import { deepMerge } from "../src";

test("Shallow merge", () => {
  const merged = deepMerge({ name: "Murilo Dutra" }, { github: "murilodutra" });
  expect(merged).toEqual({ name: "Murilo Dutra", github: "murilodutra" });
});

test("Shallow merge with overlaps", () => {
  const merged = deepMerge(
    { name: "Murilo Dutra", github: "unknown" },
    { github: "murilodutra", twitter: "murilex" }
  );
  expect(merged).toEqual({
    name: "Murilo Dutra",
    github: "murilodutra",
    twitter: "murilex",
  });
});

test("Shallow merge with arrays", () => {
  const merged = deepMerge(["vue", "react"], ["svelte", "solid"]);
  expect(merged).toEqual(["vue", "react", "svelte", "solid"]);
});

test("Deep merge with overlaps", () => {
  const merged = deepMerge(
    { name: "Murilo Dutra", accounts: { github: "unknown" } },
    { accounts: { twitter: "murilex" } }
  );
  // expect(merged).toEqual({
  //   name: "Murilo Dutra",
  //   accounts: {
  //     github: "unknown",
  //     twitter: "murilex",
  //   },
  // });
  expect(merged).toMatchSnapshot();
});

test("Throws errors on merging two different types", () => {
  expect(() => deepMerge(["foo", "bar"], { foo: "bar" })).toThrowError(
    "Error: Can not merge two different types"
  );
});
