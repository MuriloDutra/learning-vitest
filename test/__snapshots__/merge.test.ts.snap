// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Deep merge with overlaps 1`] = `
{
  "accounts": {
    "github": "unknown",
    "twitter": "murilex",
  },
  "languages": [
    "javascript",
    "typescript",
    "vue",
    "react",
  ],
  "name": "Murilo Dutra",
}
`;
